xapian-core/diversify/diversify.cc

```diff
@@ -0,0 +1,195 @@
+/** @file diversify.cc
+ *  @brief Diversification API
+ */
+/* Copyright (C) 2018 Uppinder Chugh
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian/diversify.h"
+
+#include "xapian/error.h"
+#include "debuglog.h"
+
+#include <vector>
+#include <limits>
+#include <algorithm>
+#include <cmath>
```

Comment:
It's a Xapian (and wider) coding convention to sort entries within a block of header includes - it makes it easier to see at a glance if a header is being included, and that in turn helps to avoid pointlessly including the same header more than once.

Reply:
I didn't know this, thanks for pointing this out. Fixed in new commit.